### ISSUE TITLE
Allow creating user accounts to post comments only

### ISSUE THREAD

**Issue by fflorent** [Chocobozzz/PeerTube#782]
* **PeerTube version or URL:** 1.0.0b9

On Youtube, posting comments is a key feature. On Peertube, currently it is hard to post a new comment for a new comer when the instance is close (i.e. does not allow anyone to upload videos on it). A way to achieve it would consist in creating an account in another instance or posting on Mastodon. That's hard to understand for people not aware on how Peertube works :).

What would you think of allowing creating accounts just to post comments on a video?

**Comment by Chocobozzz** [comment]
You can create a user with a video quota of 0 :)

**Comment by mcb2003** [comment]
An idea could be then: to replace the upload quota with a message stating that this account can only be used for commenting, if the quota is 0.

**Comment by Chocobozzz** [comment]
It's a good idea @mcb2003. Maybe a "Cannot upload video" would be better since a user can like/dislike videos too :)

PR welcome!

**Comment by fflorent** [comment]
Yeah, probably a user with a video quota of 0 is the idea, some thoughts about discoverability:
 - people managing a Peertube instance are not aware of this (at least, I was not ¯\_(ツ)_/¯);
 - when they set this quota, not sure whether it is clear for the users they cannot upload a video, their account can just comment;

For the first point, @mcb2003's idea is good. For the second, replacing this message in the create account panel

> Initial video quota: 5 GB

With something explaining that the account is for commenting exclusively would solve the issue :) (if that's not the case yet).

Also a question: if the quota is set to 0, is the upload button available?

Florent

**Comment by tcitworld** [comment]
In my opinion, this is a really bad idea from a user story point of view : people will want to create accounts everywhere they want to comment, which will lead to lots of useless duplicates across the fediverse.

My solution would be linked to https://github.com/Chocobozzz/PeerTube/issues/470, allowing users to view videos (and their comments) from any source - even if not allowed by the admin - into their own PeerTube instance. Users would be redirected from the external instance to their own through web protocol handlers (like Mastodon has `web+mastodon://`), and the link would carry video and maybe comment information for the own instance to find if necessary.

Therefore people would write comments with their own account on the instance they registered in the first place.

**Comment by fflorent** [comment]
@Chocobozzz I think that allowing the creation of accounts with a 0Mb quota is confusing (v 1.4.1): 

![image](https://user-images.githubusercontent.com/371705/69030501-0fbf4280-09d8-11ea-9ea2-dbbf048af2f7.png)

People are told that they can create a channel despite having no space.

**Comment by thibaultamartin** [comment]
> My solution would be linked to #470, allowing users to view videos (and their comments) from any source - even if not allowed by the admin - into their own PeerTube instance

This makes a lot of sense to me, and is a part of what I find confusing with PeerTube.
If I want to enjoy videos from several instances and what to stay "in the PeerTube world", I need one account per instance, or to find an instance following all those of interest for me.

The whole paradox is I can subscribe to several PeerTube channels on several PeerTube instances from a single Mastodon account, but I cannot do the same from a single PeerTube account.

**Comment by realityfabric** [comment]
I think that an account with a quota of 0 would work, if the interface is clear about it.

- It should be clear when signing up that the account will be for Comments Only
- Upload button should not be visible and Upload page should not be accessible (as questioned by @fflorent )
- Account should not be able to create channels
  - This would make it impossible to have public playlists as they are currently implemented

I think it's worth mentioning that YouTube does not allow people without video channels to comment or create playlists (except for the Watch Later playlist). YouTube's "no videos" accounts only allow:
- Watch Later
- Subscribe
- History

> this is a really bad idea from a user story point of view : people will want to create accounts everywhere they want to comment, which will lead to lots of useless duplicates across the fediverse.

Accounts can already comment on remote instances if their instance is following it. PeerTube instance federation is a different issue, which could in the short term be helped by users talking to their admins about following more instances. (If your admin refuses to follow those instances, then you'd have to make a new account anyway.)

>  allowing users to view videos (and their comments) from any source - even if not allowed by the admin - into their own PeerTube instance. Users would be redirected from the external instance to their own through web protocol handlers (like Mastodon has web+mastodon://), and the link would carry video and maybe comment information for the own instance to find if necessary.

As long as "even if not allowed by the admin" doesn't include "explicitly denied by the admin" this sounds great! PeerTube already allows you to open videos in Mastodon/Pleroma to comment. I'm kinda surprised it doesn't do the same with PeerTube, although PeerTube federation does work a bit differently. This sounds like something that needs to be in a different issue though.

**Comment by thibaultamartin** [comment]
A bit more details, from [this Mastodon thread (FR)](https://mamot.fr/@thibaultamartin/103210439848707517).

TL;DR: it is possible to follow remote PeerTube channels à la Mastodon. Actually, when using the search bar it is possible to search @channel@instance to follow that channel from my local account _if my local instance allows it_.

It seems to be disabled by default though. And that feature discoverability could be improved:
* By using an "autocomplete dropdown" in the search bar with example accounts from remote instances
* By replacing the "Upload" button when a user is not connected by an "Advanced search" button leading to a search page. That page would give more details regarding the possibility (or not) to follow remote channels

**Comment by kontrollanten** [comment]
I've created a PR to solve this. It's just 15 lines so I ask everyone who's interested in this to review the PR https://github.com/Chocobozzz/PeerTube/pull/3393 Thanks!
